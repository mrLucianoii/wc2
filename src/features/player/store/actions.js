import types from './constants';

function markInAction(milliseconds){
  return {
    type: types.MARK_IN,
    timeStamp: Math.round(milliseconds)
  }
}

function markOutAction(milliseconds){
  return {
    type: types.MARK_OUT,
    timeStamp: Math.round(milliseconds)
  }
}

function startPlaying(videoID) { // event passed onReady at component render
    return {
        type: types.PLAY_VIDEO,
        vidID: videoID
      }
  }
  
export default {
    markInAction,
    markOutAction,
    startPlaying,
}