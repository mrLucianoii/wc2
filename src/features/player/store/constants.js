const PLAY_VIDEO = 'weCliques-app/movies/PLAY_VIDEO';
const MOVIE_PARAM = 'weCliques-app/movies/MOVIE_PARAM';
const FETCHED_MOVIE_PROPS = 'weCliques-app/movies/FETCHED_MOVIE_PROPS';
const MARK_IN = 'weCliques-app/movies/MARK_IN';
const MARK_OUT = 'weCliques-app/movies/MARK_OUT';
const CREATED_SEGMENT = 'weCliques-app/movies/CREATED_SEGMENT';
const SEGMENT_SELECTED = 'weCliques-app/movies/SEGMENT_SELECTED';
const SEGMENT_ACTIVE = 'weCliques-app/movies/SEGMENT_ACTIVE';
const CHANGE_SEGMENT = 'weCliques-app/movies/CHANGE_SEGMENT';
const FULL_CLIP = 'weCliques-app/movies/FULL_CLIP';


export default {
    PLAY_VIDEO,
    MOVIE_PARAM,
    FETCHED_MOVIE_PROPS,
    MARK_IN,
    MARK_OUT,
    CREATED_SEGMENT,
    SEGMENT_SELECTED,
    SEGMENT_ACTIVE,
    CHANGE_SEGMENT,
    FULL_CLIP,
}