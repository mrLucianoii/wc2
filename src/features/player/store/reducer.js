import types from './constants';

const initialState = {
    selectedMovie: 'ELeMaP8EPAA',
    segmentStart: 0,
    segmentEnd: null,
};

export default (state = initialState, actions = {}) => {
  switch (actions.type) {
    case types.MARK_IN: {
      return {
        ...state,
        segmentStart: actions.timeStamp,
      }
    }
    default:

      return state;
  }
};
