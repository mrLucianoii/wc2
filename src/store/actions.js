const isLoading = () => (dispatch) => {
  dispatch({
    type: 'MOVIE_FINISHED',
  });
}; 

export default {
 isLoading,
};
