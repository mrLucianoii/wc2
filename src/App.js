import React, { Component } from 'react';
import { connect } from 'react-redux';
import { isLoadingAction as actions } from './store/actions';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  componentDidMount(){
    const { loaderAction } = this.props;
    loaderAction();
  }
  render() {
    const { loading } = this.props;
    return (
      <div className="App">
      {!loading && <h1>Loading...</h1>}
       {loading && <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          WeCliques say's, "Hello World"
        </header>}
      </div>
    );
  }
}

export default connect(
  (state) => ({
    loading: state.featureReducer.isLoading,
  }),
  (dispatch) => ({
    loaderAction: () => dispatch(actions()),
  })
)(App);
