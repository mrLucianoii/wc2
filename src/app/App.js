import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { isLoadingAction as actions } from "../store/actions";
import Player from "../features/player/Player";
import logo from "./logo.svg";
import "./App.css";
import "./index.css";

class App extends Component {
  componentDidMount() {
    const { loaderAction } = this.props;
    loaderAction();
  }
  render() {
    const { loading } = this.props;
    return (
      <div className="App">
        {!loading && <h1>Loading...</h1>}
        {loading && (
          <Fragment>
            <header className="App-header">
              <img src={logo} className="App-logo" alt="logo" />
              WeCliques say's, "Hello World"
              <Player />
            </header>
          </Fragment>
        )}
      </div>
    );
  }
}

export default connect(
  state => ({
    loading: state.featureReducer.isLoading
  }),
  dispatch => ({
    loaderAction: () => dispatch(actions())
  })
)(App);
