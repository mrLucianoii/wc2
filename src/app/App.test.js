import React from "react";
import ReactDOM from "react-dom";
import { ThemeProvider } from "react-jss";
import { Provider } from "react-redux";
import store from "../store";
import App from "./App";
import renderer from "react-test-renderer";
import theme from "./themeStyles";

const styles = () => ({
  ...theme
});

describe("App's core rendering tests", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Provider store={store()}>
        <ThemeProvider theme={styles}>
          <App />
        </ThemeProvider>
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders correctly :)", () => {
    const tree = renderer
      .create(
        <Provider store={store()}>
          <ThemeProvider theme={styles}>
            <App />
          </ThemeProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
