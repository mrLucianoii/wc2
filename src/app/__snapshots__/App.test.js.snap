// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App's core rendering tests Renders correctly :) 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    WeCliques say's, "Hello World"
    <div
      style={
        Object {
          "height": "auto",
          "width": 660,
        }
      }
    >
      <div
        className=""
      >
        <div
          className={null}
          id={null}
        />
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Play
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Pause
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Segment Start
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        In-Mark
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Out-Mark
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        full-play
      </button>
    </div>
  </header>
</div>
`;
